spec/PeriodicBackgroundSync-index.bs

```diff
@@ -92,3 +94,60 @@ Fetch requests within the <code>periodicsync</code> event while [=in the backgro
 
 ## History Leaking ## {#history-leaking}
 Fetch requests within the <code>periodicsync</code> event while [=in the background=] may reveal something about the client's navigation history to middleboxes on networks different from the one used to create the [=periodicsync registration=]. For instance, the client might visit site https://example.com, which registers a <code>periodicsync</code> event, but based on the implementation, might not fire until after the user has navigated away from the page and changed networks. Middleboxes on the new network may see the fetch requests that the <code>periodicsync</code> event makes. The fetch requests are HTTPS so the request contents will not be leaked but the domain may be (via DNS lookups and IP address of the request). To prevent this leakage of browsing history, the user agent MAY choose to only fire <code>periodicsync</code> events on the network the [=periodicsync registration=] was made on, with the understanding that it will reduce usability by not allowing synchronization opportunistically.
+
+API Description {#api-description}
+===================================
+## Extensions to the {{ServiceWorkerGlobalScope}} interface ## {#serviceworkerglobalscope-extension}
+<script type="idl">
+partial interface ServiceWorkerGlobalScope {
+    attribute EventHandler onperiodicsync;
+};
+</script>
+
+## Extensions to the {{ServiceWorkerRegistration}} interface## {#service-worker-extensions}
+
+<script type="idl">
+[Exposed=(Window,Worker)]
+partial interface ServiceWorkerRegistration {
+  readonly attribute PeriodicSyncManager periodicSync;
+};
+</script>
+
+A {{ServiceWorkerRegistration}} has a <dfn>periodic sync manager</dfn> (a {{PeriodicSyncManager}}), initially a new {{PeriodicSyncManager}} whose [=service worker registration=] is the [=context object=]'s [=service worker registration=]. The <code>periodicSync</code> attribute's getter must return the [=context object=]'s [=periodic sync manager=].
```

Comment:
It's a bit confusing that you define all of the following in a single block:

1. A PSM is owned by a SWR
    1. Initially created through ~ `new PSM(swr = this.[swr])`
1. Calling SWR.`periodicSync` return `this.[psm]`.

Can we split out (2), and avoid repetition of `context object` w/ two different values in the same paragraph?

Reply:
Split it out in a different paragraph to reduce confusion. Not sure how to avoid repetition however. Should I assign different variable names? Background Fetch spec has similar wording when introducing backgroundFetch.

Reply:
Repetition isn't per se bad, but when a definion is used twice in the same paragraph with different meanings then it's a good sign that splitting it out would improve clarity.

Reply:
Ack.